.github/workflows/ci.yml

```diff
@@ -0,0 +1,247 @@
+name: ci
+
+on:
+  workflow_call:
+    inputs:
+      # enabled by default, required for merge through default GH check
+      test_ci:
+        description: 'Enable Unit tests'
+        default: true
+        required: false
+        type: boolean
+      # disabled by default, but required for merge, there are two GH checks:
+      # - Non-Application Server integration tests
+      # - Application Server integration tests
+      # opt-in with 'ci:agent-integration'
+      agent_integration_tests_ci:
+        description: 'Enable Agent Integration tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:jdk-compatibility' tag on PR
+      jdk_compatibility_ci:
+        description: 'Enable JDK compatibility tests'
+        default: false
+        required: false
+        type: boolean
+      # disabled by default, not required for merge
+      # opt-in with 'ci:windows' tag on PR
+      windows_ci:
+        description: 'Enable Windows build & tests'
+        default: false
+        required: false
+        type: boolean
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled
+      - ready_for_review
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+env:
+  JAVA_VERSION: 17
+  JAVA_DIST: temurin
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/build
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+      - uses: ./.github/workflows/stash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+
+  license:
+    name: License
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/unstash
+        with:
+          name: build
+          path: ${{ github.workspace }}
+      - name: mvn license
+        uses: ./.github/workflows/build
+        with:
+          command: ./mvnw org.codehaus.mojo:license-maven-plugin:aggregate-third-party-report -Dlicense.excludedGroups=^co\\.elastic\\.
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+
+  unit-tests:
+    name: Unit Tests
+    if: inputs.test_ci == true || inputs.test_ci == null
```

Comment:
when does `inputs.test_ci == null` happen? Probably a comment might help for the future 🙏